src/components/Trade/SelectAsset.tsx

```diff
@@ -9,11 +9,11 @@ import { TradeRoutePaths } from './types'
 import { WithBackButton } from './WithBackButton'
 
 type SelectAssetProps = {
+  assets: Asset[]
```

Comment:
A few issues with this:

1. This is not used by parents. It's only selected, then passed down. Why drill it down then?
2. As a general rule, it's better to use normalized data if possible - I know this iterates on top of the existing `filterBy` which was removed and used to be assets in, assets out, but IDs seem safer. If we then select assets by ID, that would be a simple O(1) lookup to retrieve an asset
3. While selectors themselves and their references are stable (well until they're referentially invalidated because of input or output changes obviously), react uses shallow comparison for comparison so that makes this potentially unstable

Reply:
I've discussed this with gomes just now. Intent is to sidestep unmount behaviour of chakra modals where we need to maintain memoized filtered assets for the lifetime of the parent component rather than the time a modal is open.

Resolution here is to keep the ability to pass in assets, with default behaviour to select all assets. I'll make the `assets` prop optional and handle the default case internal to `AssetSearch`

Reply:
Yep, as discussed, this one definitely makes sense from a performance point of view as well as architecturally. `AssetSearch />` is the one that could benefit from sane defaults and no props drilling!